codec/src/main/java/io/netty/handler/codec/compression/JdkZlibDecoder.java

```diff
@@ -280,6 +284,15 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
         }
     }
 
+    private int inflate(ByteBuf decompressed, int writerIndex, int writable) throws DataFormatException {
+        if (IS_JAVA_11_INFLATER) {
+            return Java11ZlibUtils.inflate(inflater, decompressed.internalNioBuffer(writerIndex, writable));
```

Comment:
We have an opportunity to use `ctx.alloc().buffer(preferredSize)` instead of `ctx.alloc().heapBuffer(preferredSize)` for `decompressed` if `IS_JAVA_11_INFLATER` is available.